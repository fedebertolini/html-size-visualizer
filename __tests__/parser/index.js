const parser = require("../../parser");

describe("parser", () => {
  it("should parse the simple html file", () => {
    const filePath = process.cwd() + "/__tests__/parser/files/simple.html";

    const domTree = parser.parseFile(filePath);
    expect(domTree).toMatchSnapshot();
  });
});
