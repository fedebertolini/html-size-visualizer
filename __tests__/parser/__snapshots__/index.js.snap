// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser should parse the simple html file 1`] = `
Object {
  "attributes": Object {},
  "children": Array [
    Object {
      "attributes": Object {
        "lang": "en",
      },
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [],
          "estimatedSize": 26,
          "nodeType": "tag",
          "tagName": "title",
        },
        Object {
          "attributes": Object {
            "content": "This is a simple HTML document",
            "name": "description",
          },
          "children": Array [],
          "estimatedSize": 73,
          "nodeType": "tag",
          "tagName": "meta",
        },
      ],
      "estimatedSize": 122,
      "nodeType": "tag",
      "tagName": "head",
    },
    Object {
      "attributes": Object {},
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [],
          "estimatedSize": 20,
          "nodeType": "tag",
          "tagName": "h1",
        },
        Object {
          "attributes": Object {},
          "children": Array [],
          "estimatedSize": 37,
          "nodeType": "tag",
          "tagName": "p",
        },
      ],
      "estimatedSize": 85,
      "nodeType": "tag",
      "tagName": "body",
    },
  ],
  "estimatedSize": 220,
  "nodeType": "tag",
  "tagName": "html",
}
`;
